docs/DeveloperGuide.md

```diff
@@ -237,6 +237,70 @@ The following activity diagram summarizes what happens when a user executes a ne
   * Cons: We must ensure that the implementation of each individual command are correct.
 
 
+### [Proposed] Command History
+
+#### Implementation
+The proposed commandHistory mechanism is facilitated by `CommandHistory`. It contains `commandHistoryList` and `currentCommandPointer`. Additionally, it implements the following operations:
+* `CommandHistory#hasPreviousCommand()` — Returns if there is a previously command from the history based on `currentCommandPointer`.
+* `CommandHistory#hasNextCommand()` — Returns if there is a next command from the history based on `currentCommandPointer`.
+* `CommandHistory#getPreviousCommand()` — Restores the previous command from its history based on `currentCommandPointer`.
+* `CommandHistory#getNextCommand()` — Restores the next command from its history based on `currentCommandPointer`.
+* `CommandHistory#addCommand()` — Add the command into the history.
+
+
+Given below is an example usage scenario and how the commandHistory behaves at each step.
+
+Step 1. The user launches the application for the first time and enters their first command. 
+The `CommandHistory` will save the command and the `currentCommandPointer` won't be pointing to any command.
+
+![CommandHistoryState0](images/CommandHistoryState0.png)
+
+Step 2. The user presses '↑' while the commandBox is selected. `CommandHistory#getPreviousCommand()` is called and the previous command is displayed in the commandBox. 
+When the previous command is entered into the commandBox, the newly edited version of command will not be stored in `CommandHistory`.
+
+![CommandHistoryState1](images/CommandHistoryState1.png)
+
+Step 3. The user presses '↑' while the commandBox is selected. `CommandHistory#getPreviousCommand()` is called 
+and the previous command is displayed in the commandBox. When the previous command is edited and is entered into the commandBox, the newly edited version of command will be stored in `CommandHistory` after calling `CommandHistory#addCommand`.
```

Comment:
`CommandHistory#addCommand` should be `CommandHistory#addCommand()`.

Reply:
Thanks I have fixed this in https://github.com/AY2324S1-CS2103T-F11-1/tp/pull/163/commits/0777b776c9cd2502fa380f09ee4e11f0d1450543